napari/_qt/widgets/qt_viewer_dock_widget.py

```diff
@@ -316,19 +337,24 @@ def __init__(
         line = QFrame(self)
         line.setObjectName("QtCustomTitleBarLine")
 
-        self.hide_button = QPushButton(self)
-        self.hide_button.setToolTip(trans._('hide this panel'))
-        self.hide_button.setObjectName("QTitleBarHideButton")
-        self.hide_button.setCursor(Qt.CursorShape.ArrowCursor)
-        self.hide_button.clicked.connect(lambda: self.parent().close())
-
-        self.float_button = QPushButton(self)
-        self.float_button.setToolTip(trans._('float this panel'))
-        self.float_button.setObjectName("QTitleBarFloatButton")
-        self.float_button.setCursor(Qt.CursorShape.ArrowCursor)
-        self.float_button.clicked.connect(
-            lambda: self.parent().setFloating(not self.parent().isFloating())
-        )
+        if hide_btn:
+            self.hide_button = QPushButton(self)
+            self.hide_button.setToolTip(trans._('hide this panel'))
+            self.hide_button.setObjectName("QTitleBarHideButton")
+            self.hide_button.setCursor(Qt.CursorShape.ArrowCursor)
+            self.hide_button.clicked.connect(lambda: self.parent().close())
+
+        if float_btn:
+            self.float_button = QPushButton(self)
+            self.float_button.setToolTip(trans._('float this panel'))
+            self.float_button.setObjectName("QTitleBarFloatButton")
+            self.float_button.setCursor(Qt.CursorShape.ArrowCursor)
+            self.float_button.clicked.connect(
+                lambda: self.parent().setFloating(
+                    not self.parent().isFloating()
+                )
+            )
```

Comment:
Hm. I prefer to control visibility, not the presence of buttons. Control presence breaks widget API.

Reply:
Would it be okay to instead control the presence of the custom title bar?